autorest/adal/token.go

```diff
@@ -932,9 +932,9 @@ func (spt *ServicePrincipalToken) refreshInternal(ctx context.Context, resource
 	req = req.WithContext(ctx)
 	var resp *http.Response
 	if msiSecret, ok := spt.inner.Secret.(*ServicePrincipalMSISecret); ok {
-		req.Method = http.MethodGet
 		switch msiSecret.msiType {
 		case msiTypeAppServiceV20170901:
+			req.Method = http.MethodGet
 			req.Header.Set("secret", os.Getenv(msiSecretEnv))
 			break
```

Comment:
Does this env not add the clientid or am I missing it somewhere?

Reply:
That happens in `newServicePrincipalTokenFromMSI()`.  In that function the endpoint is created.  Then in `refreshInternal()` the HTTP request is created.  So the complete request creation is split over two places :(

Reply:
Ah I see...that threw me off for a second there. So refreshInternal is called even for the first request?

Reply:
Correct.